GoogleChromeLabs/ps-analysis-tool
# CLI: Check for latest version and show message

## Description

Add version check to the PSAT CLI to notify users when they are running an outdated version. Also, add the version is included in all generated reports for CLI and extension.

## Relevant Technical Choices

- Use the `update-notifier` package to check for the latest version and notify users.
- Use the package version from `package.json` instead of hardcoding it.
- Pass the version to the report and use it in the header component.
- Retrieve the current extension version and include it in the header of the extension report.
- Add `psatVersion` in downloaded report JSON file.

## Testing Instructions

- Go to `packages/cli/package.json` and reduce the version number.
- Build and run the CLI to test if it can check and notify about the update.
- The latest version check occurs after a certain time (e.g., 1 day) instead of running every time to reduce its impact on time and performance. This is currently disabled for easier QA testing. Once QA is complete, we'll enable it for further testing.
- Open the dashboard report and verify that the current CLI version appears in the report header.
- Download a report from the extension and confirm that the current extension version is displayed (which may differ from the CLI version).
- Download report from the CLI dashboard and check if it includes `psatVersion` key in JSON file.

## Additional Information:

## Screenshot/Screencast

![image](https://github.com/user-attachments/assets/80e5b597-341c-4995-ad36-7a47b513c91c)

---

## Checklist






Fixes #815
